shoes-swt/lib/shoes/swt/common/painter.rb

```diff
@@ -33,18 +33,18 @@ def paint_object(graphics_context)
           cap = LINECAP[@obj.dsl.style[:cap]]
           graphics_context.set_line_cap(cap) if cap
           graphics_context.set_transform(@obj.transform)
+
           obj = @obj.dsl
-          case obj
-            when ::Shoes::Oval, ::Shoes::Rect
-              set_rotate graphics_context, obj.rotate,
-                         obj.element_left + obj.element_width / 2.0,
-                         obj.element_top + obj.element_height / 2.0 do
-                fill graphics_context if fill_setup(graphics_context)
-                draw graphics_context if draw_setup(graphics_context)
-              end
-            else
+          if needs_rotate?
+            set_rotate graphics_context, obj.rotate,
+                       obj.element_left + obj.element_width / 2.0,
+                       obj.element_top + obj.element_height / 2.0 do
               fill graphics_context if fill_setup(graphics_context)
               draw graphics_context if draw_setup(graphics_context)
+            end
+          else
+            fill graphics_context if fill_setup(graphics_context)
+            draw graphics_context if draw_setup(graphics_context)
```

Comment:
Yeah, let me look at flattening that out a bit more while I'm here. It takes a little state since we have to undo the transform afterward, but might be cleaner.